### PR TITLE
Added an isIterable predicate

### PR DESCRIPTION
**Pull request by dalefrancis88**
This is a PR to add an `isIterable` predicate in preparation for work on #29

This is something we'll obviously need internally, however I am unsure on the value of adding this as something that is part of the api, thoughts?

### CODE REVIEW

File: src/predicates/isIterable.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Dale Francis (dalefrancis88) */
+
+module.exports =
+  require('../core/isPromise')
```
Comment (evilsoft) [inline src/predicates/isIterable.js]: This is including `isPromise`

File: src/predicates/isIterable.spec.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const test = require('tape')
+
+const isFunction = require('../core/isFunction')
+
+const isMonad = require('./isIterable')
```
Comment (evilsoft) [inline src/predicates/isIterable.spec.js]: this should be named `isIterable`

File: src/core/isIterable.spec.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  t.equal(isIterable(undefined), false, 'returns false for undefined')
+  t.equal(isIterable(null), false, 'returns false for null')
+  t.equal(isIterable(0), false, 'returns false for falsey number')
+  t.equal(isIterable(1), false, 'returns false for truthy number')
+  t.equal(isIterable(false), false, 'returns false for false')
+  t.equal(isIterable(true), false, 'returns false for true')
+  t.equal(isIterable({}), false, 'returns false for an object')
+  t.equal(isIterable(identity), false, 'returns false for function')
+
+  t.equal(isIterable([]), true, 'returns false for an array')
```
Comment (evilsoft) [inline src/core/isIterable.spec.js]: text should read 'returns *true* for an array'

File: src/core/isIterable.spec.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  t.equal(isIterable('string'), true, 'returns true for truthy string')
```
Comment (evilsoft) [inline src/core/isIterable.spec.js]: We should also add a test for falsey string, just to be safe.

File: src/core/isIterable.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Dale Francis (dalefrancis88) */
+
+const isFunction = require('./isFunction')
+
+function isIterable(iterable) {
+  return iterable !== null && iterable !== undefined && isFunction(iterable[Symbol.iterator])
```
Comment (evilsoft) [inline src/core/isIterable.js]: with `NaN` also not being iterable, could we use the `isNil` predicate here?

File: src/predicates/isIterable.js (hunk 1 above)
Comment (dalefrancis88) [reply src/predicates/isIterable.js]: That's what i get for copying, fail :)

File: src/predicates/isIterable.spec.js (hunk 2 above)
Comment (dalefrancis88) [reply src/predicates/isIterable.spec.js]: another fail copy :/

File: src/core/isIterable.spec.js (hunk 3 above)
Comment (dalefrancis88) [reply src/core/isIterable.spec.js]: ok i need to stop copying :/

File: src/core/isIterable.js (hunk 5 above)
Comment (dalefrancis88) [reply src/core/isIterable.js]: cool, good idea, will do it

**Pull request merged**